packages/arb-token-bridge-ui/src/hooks/useArbTokenBridge.ts

```diff
@@ -110,21 +113,24 @@ export const useArbTokenBridge = (
   const [bridgeTokens, setBridgeTokens] = useState<
     ContractStorage<ERC20BridgeToken> | undefined
   >(undefined)
+  const [networks] = useNetworks()
+  const { childChainProvider, parentChainProvider } =
+    useNetworksRelationship(networks)
 
   const { addPendingTransaction } = useTransactionHistory(walletAddress)
 
   const {
     eth: [, updateEthL1Balance],
     erc20: [, updateErc20L1Balance]
   } = useBalance({
-    provider: l1.provider,
+    provider: parentChainProvider,
     walletAddress
   })
   const {
     eth: [, updateEthL2Balance],
     erc20: [, updateErc20L2Balance]
   } = useBalance({
-    provider: l2.provider,
+    provider: childChainProvider,
```

Comment:
l1 & l2 from params are stale so had to do this

Reply:
We should probably use that across the entire hook and get rid of it in params

Reply:
100%
@chrstph-dvx is working on that 😁